### ISSUE TITLE
File handling in Gatsby

### ISSUE THREAD

**Issue by ajayns** [collective/gatsby-source-plone#95]
Handling:
1. Images
2. All other file types

**Comment by ajayns** [comment]
I've decided it a good idea to start with handling just images first and then move on to focusing on all file types. And so in `image-handling` branch in my fork of the repo, I've made an implementation based on the docs which I could find. According to it, I should be able to query the node via `localFile` from the file node but this doesn't seem to be possible even though the project compiles fine with Gatsby. 

Currently, I'm looking into this issue, will get feedback from the Gatsby community as well. If you have any ideas on what could be going wrong, do let me know and also checkout my branch.

@datakurre @cekk

**Comment by datakurre** [comment]
@ajayns I believe that your code works as expected. It's just that we have no images in our test data. But we do have PloneNewsItems with images.

@cekk @tisto We may not get far with images and files without being able to configure (or autoconfigure) which content types have images in which fields. Good ideas are welcome! Should the plugin load content type schemas from the Plone site (with rest api) and introspect from that data?

**Comment by cekk** [comment]
Yes, could be a good point. For sure with restapi we can get all of these informations.

I'm for splitting this thing in two tasks:
- fix this issue for base content-type (File, Image and News Item) now
- Open a separate task (for the future, if we have some time) for dynamic file/image handling

i'd like to first have a fully working product with base contents, and then improve it

**Comment by datakurre** [comment]
@ajayns I tried out your code here https://github.com/collective/gatsby-source-plone/commit/fe5af457fa4719c69a0621d248630b3addc4ccdd

**Comment by datakurre** [comment]
Types-endpoint for future reference https://plonerestapi.readthedocs.io/en/latest/types.html

**Comment by ajayns** [comment]
@datakurre Great, I was checking it out properly, updated my branch and added comments explaining it. I think that manages News Items, so the next step would be to handle all other file types. I'm looking into handling that right now.

**Comment by ajayns** [comment]
@datakurre Let's create a new issue, I'll start a discussion there.

**Comment by datakurre** [comment]
@ajayns This probably still needs some documentation. It is not obvious that gatsby-source-filesystem must be  configured for publicURLs of remote file nodes to work. Possibly you could also contribute to Gatsby docs if you wish.

**Comment by ajayns** [comment]
@datakurre, yes! I will write up on that!

**Comment by ajayns** [comment]
@datakurre Think this issue can be closed now?
